src/v/kafka/server/quota_manager.cc

```diff
@@ -480,4 +536,43 @@ ss::future<> quota_manager::do_gc(clock::time_point expire_threshold) {
       });
 }
 
+void client_quotas_probe::setup_metrics() {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    auto metric_defs = std::vector<ss::metrics::impl::metric_definition_impl>{};
+    auto rule_label = ss::metrics::label("quota_rule");
+    auto quota_type_label = ss::metrics::label("quota_type");
+
+    for (auto quota_type : all_client_quota_types) {
+        for (auto rule : all_client_quota_rules) {
+            metric_defs.emplace_back(sm::make_histogram(
+              "client_quota_throttle_time",
+              [this, rule, quota_type] {
+                  return get_throttle_time(rule, quota_type);
+              },
+              sm::description("Client quota throttling delay per rule and "
+                              "quota type (in micros)"),
+              {rule_label(rule), quota_type_label(quota_type)}));
+            metric_defs.emplace_back(sm::make_histogram(
+              "client_quota_throughput",
+              [this, rule, quota_type] {
+                  return get_throughput(rule, quota_type);
+              },
+              sm::description(
+                "Client quota throughput per rule and quota type"),
+              {rule_label(rule), quota_type_label(quota_type)}));
+        }
+    }
+
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("kafka:quotas"),
+      std::move(metric_defs),
+      {sm::shard_label}, // Always aggregate across shards
+      {sm::shard_label});
```

Comment:
question: considering that these are internal-only, is it true that we never ever want to see per-shard metrics? we just have the one sharded quota_manager, so cardinality wouldn't bee too crazy, right?

Reply:
> considering that these are internal-only, is it true that we never ever want to see per-shard metrics? we just have the one sharded quota_manager, so cardinality wouldn't be too crazy, right?

Yeah, it's 1 sharded quota_manager but the label cardinality is "shard label if not aggregated" * 3 quota types * 6 quota rules * "however many histogram buckets". So I think we should aggregate the shard label away.